src/Microsoft.SqlTools.ServiceLayer/Scripting/ScriptingExtensionMethods.cs

```diff
@@ -103,10 +103,10 @@ public static Urn ToUrn(this ScriptingObject scriptingObject, string server, str
             string urn = string.Format(
                 "Server[@Name='{0}']/Database[@Name='{1}']/{2}[@Name='{3}' {4}]",
                 server.ToUpper(),
-                database,
+                Urn.EscapeString(database),
```

Comment:
How did you end up uncovering this bug? Do we have a test coverage hole? If you can spare on some thoughts what missing test we ought to go and write then kindly file an issue for our backlog.

Reply:
I see that you have added some new tests below, so if they were sufficient to reproduce the original bug and verify the fix then we are good here and thank you for the tests :)